ale_linters/matlab/mlint.vim

```diff
@@ -0,0 +1,59 @@
+" Author: awlayton <alex@layton.in>
+" Description: mlint for MATLAB files
+
+if exists('g:loaded_ale_linters_matlab_mlint')
+    finish
+endif
+
+let g:loaded_ale_linters_matlab_mlint = 1
+
+let g:ale_matlab_mlint_executable =
+\   get(g:, 'ale_matlab_mlint_executable', 'mlint')
+
+function! ale_linters#matlab#mlint#Handle(buffer, lines)
+    " Matches patterns like the following:
+    "
+    " L 27 (C 1): FNDEF: Terminate statement with semicolon to suppress output.
+    " L 30 (C 13-15): FNDEF: A quoted string is unterminated.
+    let l:pattern = '^L \(\d\+\) (C \([0-9-]\+\)): \([A-Z]\+\): \(.\+\)$'
+    let l:output = []
+
+    for l:line in a:lines
+        let l:match = matchlist(l:line, l:pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let l:lnum = l:match[1] + 0
+        let l:col = l:match[2] + 0
+        let l:code = l:match[3]
+        let l:text = l:match[4]
+
+        " Suppress erroneous waring about file name
+        if l:code ==# 'FNDEF'
```

Comment:
We might be able to turn this on in future. There is some support for copying the same filename to the temporary directory to come eventually.